clients/drcachesim/tools/schedule_stats.h

```diff
@@ -190,6 +198,37 @@ class schedule_stats_t : public analysis_tool_t {
         std::unordered_set<memref_tid_t> threads;
         std::unique_ptr<histogram_interface_t> instrs_per_switch;
     };
+
+    struct workload_tid_t {
+        workload_tid_t(int64_t workload, int64_t thread)
+            : workload_id(workload)
+            , tid(thread)
+        {
+        }
+        bool
+        operator==(const workload_tid_t &rhs) const
+        {
+            return workload_id == rhs.workload_id && tid == rhs.tid;
+        }
+
+        bool
+        operator!=(const workload_tid_t &rhs) const
+        {
+            return !(*this == rhs);
+        }
+        int64_t workload_id;
+        int64_t tid;
+    };
+
+    struct workload_tid_hash_t {
+        std::size_t
+        operator()(const workload_tid_t &wt) const
+        {
+            return std::hash<size_t>()(wt.workload_id) ^
+                std::hash<memref_tid_t>()(wt.tid);
```

Comment:
It seems like this could cause {workload_id=X, tid=Y} to have the same hash as {workload_id=Y, tid=X}.  If that's a concern, maybe do something asymmetric like std::hash<size_t>()(wt.workload_id ^ wt.tid) ^
std::hash<memref_tid_t>()(wt.tid)

Reply:
Done, with a comment explaining.